lib/chat_channel_fetcher.rb

```diff
@@ -171,22 +130,36 @@ def self.secured_direct_message_channels(user_id, memberships, guardian)
         UserField.all.pluck(:id).map { |fid| "#{User::USER_FIELD_PREFIX}#{fid}" }
     User.preload_custom_fields(channels.map { |c| c.chatable.users }.flatten, preload_fields)
 
-    unread_counts_per_channel = unread_counts(channels, user_id)
+    decorate_memberships_with_tracking_data(guardian, channels, memberships)
+  end
 
-    channels.filter_map do |channel|
-      next if !guardian.can_see_chat_channel?(channel)
+  def self.decorate_memberships_with_tracking_data(guardian, channels, memberships)
```

Comment:
Okay I am still not really happy with how this part is, trying to think of a different way of doing this. Now I've moved `unread_count` and `unread_mentions` down to the membership level as model attributes I have to decorate them still. Maybe we have a sub-section of the channel serializer for current user tracking state as well to put this? Not sure...also really not a fan that we have to parse the notification JSON data out